tests/_core/test_transformations.py

```diff
@@ -0,0 +1,361 @@
+import json
+
+import numpy as np
+import pytest
+
+from spatialdata._core.coordinate_system import CoordinateSystem
+from spatialdata._core.transformations import (
+    Affine,
+    BaseTransformation,
+    ByDimension,
+    Identity,
+    MapAxis,
+    Rotation,
+    Scale,
+    Sequence,
+    Translation,
+)
+from tests._core.conftest import (
+    c_cs,
+    cyx_cs,
+    x_cs,
+    xy_cs,
+    xyc_cs,
+    xyz_cs,
+    y_cs,
+    yx_cs,
+    z_cs,
+    zyx_cs,
+)
+
+
+def _test_transformation(
+    transformation: BaseTransformation,
+    original: np.ndarray,
+    transformed: np.ndarray,
+    input_cs: CoordinateSystem,
+    output_cs: CoordinateSystem,
+    wrong_output_cs: CoordinateSystem,
+    test_affine: bool = True,
+    test_affine_inverse: bool = True,
+    test_inverse: bool = True,
+):
+    # missing input and output coordinate systems
+    with pytest.raises(ValueError):
+        assert np.allclose(transformation.transform_points(original), transformed)
+
+    # missing output coordinate system
+    transformation.input_coordinate_system = input_cs
+    with pytest.raises(ValueError):
+        assert np.allclose(transformation.transform_points(original), transformed)
+
+    # wrong output coordinate system
+    transformation.output_coordinate_system = wrong_output_cs
+    try:
+        # if the output coordinate system still allows to compute the transformation, it will give points different
+        # from the one we expect
+        assert not np.allclose(transformation.transform_points(original), transformed)
+    except ValueError:
+        # covers the case in which the tranformation failed because of an incompatible output coordinate system
+        pass
+
+    # wrong points shapes
+    transformation.output_coordinate_system = output_cs
+    with pytest.raises(ValueError):
+        assert np.allclose(transformation.transform_points(original.ravel()), transformed.ravel())
+    with pytest.raises(ValueError):
+        assert np.allclose(transformation.transform_points(original.transpose()), transformed.transpose())
+    with pytest.raises(ValueError):
+        assert np.allclose(transformation.transform_points(np.expand_dims(original, 0)), np.expand_dims(transformed, 0))
```

Comment:
I'm not sure I understand these tests. Is the `assert np.allclose()` necessary? Should it just be 

```python
with pytest.raises(ValueError):
    transformation.transform_points(...)
```

Reply:
you are right, changed it

Reply:
pushed by accident in `napari_fix`, so gonna be merged when we merge that branch (very soon)